From d672011e10097e5e61659a5d64ac9cb7b7544b60 Mon Sep 17 00:00:00 2001
From: Linus Walleij <linus.walleij@linaro.org>
Date: Wed, 25 Dec 2024 01:09:20 +0100
Subject: [PATCH] ARM: dts: ixp4xx OpenWrt LED aliases

This outoftree patch adds OpenWrt LED aliases to the DTS files
of supported devices.

Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
---
 .../boot/dts/intel/ixp/intel-ixp42x-dlink-dsm-g600.dts |  9 ++++++---
 .../boot/dts/intel/ixp/intel-ixp42x-freecom-fsg-3.dts  | 10 +++++++---
 .../dts/intel/ixp/intel-ixp42x-gateworks-gw2348.dts    |  6 +++++-
 .../boot/dts/intel/ixp/intel-ixp42x-iomega-nas100d.dts |  9 ++++++---
 .../boot/dts/intel/ixp/intel-ixp42x-linksys-nslu2.dts  |  9 ++++++---
 .../dts/intel/ixp/intel-ixp42x-netgear-wg302v1.dts     |  3 +++
 .../dts/intel/ixp/intel-ixp42x-usrobotics-usr8200.dts  |  5 ++++-
 .../dts/intel/ixp/intel-ixp43x-gateworks-gw2358.dts    |  6 +++++-
 8 files changed, 42 insertions(+), 15 deletions(-)

--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-dlink-dsm-g600.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-dlink-dsm-g600.dts
@@ -31,16 +31,19 @@
 	};
 
 	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
 		serial0 = &uart0;
 	};
 
 	leds {
 		compatible = "gpio-leds";
-		led-power {
+		led_power: led-power {
 			label = "dsmg600:green:power";
 			gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
-			linux,default-trigger = "heartbeat";
 		};
 		led-wlan {
 			label = "dsmg600:green:wlan";
@@ -48,7 +51,7 @@
 			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
 			default-state = "on";
 			/* We don't have WLAN trigger in the kernel (yet) */
-			linux,default-trigger = "netdev";
+			linux,default-trigger = "phy0tx";
 		};
 	};
 
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-freecom-fsg-3.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-freecom-fsg-3.dts
@@ -29,6 +29,10 @@
 	};
 
 	aliases {
+		led-boot = &led_ring;
+		led-failsafe = &led_sync;
+		led-running = &led_ring;
+		led-upgrade = &led_sync;
 		serial0 = &uart0;
 	};
 
@@ -112,7 +116,7 @@
 					reg = <0x00 0x02>;
 					mask = <0x01>;
 					label = "fsg:blue:wlan";
-					linux,default-trigger = "wlan";
+					linux,default-trigger = "phy0tx";
 					default-state = "on";
 				};
 				led@0,1 {
@@ -139,7 +143,7 @@
 					linux,default-trigger = "";
 					default-state = "on";
 				};
-				led@0,4 {
+				led_sync: led@0,4 {
 					compatible = "register-bit-led";
 					reg = <0x00 0x02>;
 					mask = <0x08>;
@@ -147,7 +151,7 @@
 					linux,default-trigger = "";
 					default-state = "on";
 				};
-				led@0,5 {
+				led_ring: led@0,5 {
 					compatible = "register-bit-led";
 					reg = <0x00 0x02>;
 					mask = <0x10>;
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-gateworks-gw2348.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-gateworks-gw2348.dts
@@ -26,12 +26,16 @@
 	};
 
 	aliases {
+		led-boot = &usr_led;
+		led-failsafe = &usr_led;
+		led-running = &usr_led;
+		led-upgrade = &usr_led;
 		serial0 = &uart0;
 	};
 
 	leds {
 		compatible = "gpio-leds";
-		led-user {
+		usr_led: led-user {
 			label = "gw2348:green:user";
 			gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
 			default-state = "on";
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-iomega-nas100d.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-iomega-nas100d.dts
@@ -26,6 +26,10 @@
 	};
 
 	aliases {
+		led-boot = &pwr_led;
+		led-failsafe = &pwr_led;
+		led-running = &pwr_led;
+		led-upgrade = &pwr_led;
 		serial0 = &uart0;
 	};
 
@@ -36,7 +40,7 @@
 			gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
 			/* We don't have WLAN trigger in the kernel (yet) */
-			linux,default-trigger = "netdev";
+			linux,default-trigger = "phy0tx";
 		};
 		led-disk {
 			label = "nas100d:red:disk";
@@ -44,11 +48,10 @@
 			default-state = "on";
 			linux,default-trigger = "disk-activity";
 		};
-		led-power {
+		pwr_led: led-power {
 			label = "nas100d:red:power";
 			gpios = <&gpio0 15 GPIO_ACTIVE_LOW>;
 			default-state = "on";
-			linux,default-trigger = "heartbeat";
 		};
 	};
 
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-linksys-nslu2.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-linksys-nslu2.dts
@@ -26,18 +26,21 @@
 	};
 
 	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_ready;
+		led-upgrade = &led_status;
 		serial0 = &uart0;
 	};
 
 	leds {
 		compatible = "gpio-leds";
-		led-status {
+		led_status: led-status {
 			label = "nslu2:red:status";
 			gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
-			linux,default-trigger = "heartbeat";
 		};
-		led-ready {
+		led_ready: led-ready {
 			label = "nslu2:green:ready";
 			gpios = <&gpio0 1 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-netgear-wg302v1.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-netgear-wg302v1.dts
@@ -29,6 +29,9 @@
 	};
 
 	aliases {
+		led-boot = &test_led;
+		led-failsafe = &test_led;
+		led-upgrade = &test_led;
 		/* These are switched around */
 		serial0 = &uart1;
 	};
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp42x-usrobotics-usr8200.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp42x-usrobotics-usr8200.dts
@@ -29,6 +29,10 @@
 	};
 
 	aliases {
+		led-boot = &pwr_led;
+		led-failsafe = &pwr_led;
+		led-running = &pwr_led;
+		led-upgrade = &pwr_led;
 		/* These are switched around */
 		serial0 = &uart1;
 		serial1 = &uart0;
@@ -67,7 +71,6 @@
 			label = "usr8200:green:pwr";
 			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
-			linux,default-trigger = "heartbeat";
 		};
 	};
 
--- a/arch/arm/boot/dts/intel/ixp/intel-ixp43x-gateworks-gw2358.dts
+++ b/arch/arm/boot/dts/intel/ixp/intel-ixp43x-gateworks-gw2358.dts
@@ -25,12 +25,16 @@
 	};
 
 	aliases {
+		led-boot = &usr_led;
+		led-failsafe = &usr_led;
+		led-running = &usr_led;
+		led-upgrade = &usr_led;
 		serial0 = &uart0;
 	};
 
 	leds {
 		compatible = "gpio-leds";
-		led-user {
+		usr_led: led-user {
 			label = "gw2358:green:LED";
 			gpios = <&pld1 0 GPIO_ACTIVE_LOW>;
 			default-state = "on";
